### PR TITLE
Brainx Bid Adapter : initial release

### PR DESCRIPTION
**Pull request by Hugh0222**
## Type of change

- [x] Bugfix
- [ ] Feature
- [x] New bidder adapter  
- [x] Updated bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

### CODE REVIEW

File: modules/brainxBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  buildRequests(bidRequests, bidderRequest) {
+    const data = converter.toORTB({ bidRequests, bidderRequest })
+    const device = config.getConfig('device') || {};
+    // console.log('data-==========', data);
+    // console.log('bidRequests-==========', bidRequests);
+    // console.log('bidderRequest-==========', bidderRequest);
+    data.user = {
+      buyeruid: generateUUID()
+    }
+    data.device.ip = navigator.ip || '202.100.48.46';
```
Comment (patmmccann) [inline modules/brainxBidAdapter.js:52]: what is this fallback lol :)

File: modules/brainxBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    return true;
+  },
+  buildRequests(bidRequests, bidderRequest) {
+    const data = converter.toORTB({ bidRequests, bidderRequest })
+    const device = config.getConfig('device') || {};
+    // console.log('data-==========', data);
+    // console.log('bidRequests-==========', bidRequests);
+    // console.log('bidderRequest-==========', bidderRequest);
+    data.user = {
+      buyeruid: generateUUID()
```
Comment (patmmccann) [inline modules/brainxBidAdapter.js:50]: this is odd, it will change every request

**Comment by Hugh0222** [comment]
Our current processing is temporary, and the main purpose is to determine
what problems there are during the review process. Regarding the fixed
writing method, we are already adjusting it~

File: modules/brainxBidAdapter.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    return true;
+  },
+  buildRequests(bidRequests, bidderRequest) {
+    const data = converter.toORTB({ bidRequests, bidderRequest })
+    data.user = {
+      buyeruid: generateUUID()
+    }
+    return {
+      method: METHOD,
+      url: `${String(deepAccess(bidRequests[0], 'params.endpoint'))}?token=${String(deepAccess(bidRequests[0], 'params.pubId'))}`,
```
[the remark this answers is not in the document]
Comment (Hugh0222) [reply modules/brainxBidAdapter.js:50]: OK, thank you for your reply. We will continue to make adjustments

[the remark this answers is not in the document]
Comment (Hugh0222) [reply modules/brainxBidAdapter.js]: Hello, bretg, we have made adjustments to the above issues. Please review it again.

**Review by bretg — APPROVED** [review]
my review only covers that endpoint is no longer a required param

**Comment by bretg** [comment]
docs PR https://github.com/prebid/prebid.github.io/pull/5689

**Comment by Hugh0222** [comment]
> 文档 PR [prebid/prebid.github.io#5689](https://github.com/prebid/prebid.github.io/pull/5689)

Hi bretg, I was wondering when the next merge code will be released?

**Comment by Hugh0222** [comment]
> 我的评论仅涵盖端点不再是必需参数

Hi bretg, I was wondering when the next merge code will be released?

**Comment by bretg** [comment]
@ChrisHuie - please assign this to someone for proper review. 

@Hugh0222 - sorry, we do not have a committed service-level agreement on how long it takes a pull request to get through review. It depends on the complexity and quality of the code as well as the availability of qualified reviewers.

File: integrationExamples/gpt/hello_world.html [hunk 4]
```diff
@@ -20,15 +20,15 @@
...
             },
 
             // Replace this object to test a new Adapter!
             bids: [{
-                bidder: 'appnexus',
+                bidder: 'brianx',
                 params: {
-                    placementId: 13144370
+                    pubId: 'F7B53DBC-85C1-4685-9A06-9CF4B6261FA3'
                 }
```
Comment (ChrisHuie) [inline integrationExamples/gpt/hello_world.html:32]: please remove changes to this file in pr

File: integrationExamples/gpt/x-domain/creative.html [hunk 5]
```diff
@@ -2,7 +2,7 @@
...
 
 <script>
     pbRender({
```
Comment (ChrisHuie) [inline integrationExamples/gpt/x-domain/creative.html:8]: please remove changes to this file in the pr

File: integrationExamples/gpt/hello_world.html (hunk 4 above)
Comment (Hugh0222) [reply integrationExamples/gpt/hello_world.html]: Already restored

File: integrationExamples/gpt/x-domain/creative.html (hunk 5 above)
Comment (Hugh0222) [reply integrationExamples/gpt/x-domain/creative.html]: Already restored

**Review by ChrisHuie — APPROVED** [review]

**Pull request merged**